ice.c

```diff
@@ -1225,6 +1225,7 @@ void janus_ice_component_destroy(janus_ice_component *component) {
 		janus_refcount_decrease(&stream->ref);
 		component->stream = NULL;
 	}
+	janus_dtls_srtp_destroy(component->dtls);
```

Comment:
Won't this cause problems? We normally do this in `janus_ice_component_free`, where we also set the pointer to NULL, though. Since you're not doing it here, the references may have gone to 0 when we're about to free the component, and the further call to `janus_dtls_srtp_destroy` might crash it.

Reply:
`janus_ice_component_free` is only called when the the refcount of the component reaches zero - which now only happens when the sctp thread ends. By calling `janus_dtls_srtp_destroy` from `janus_ice_component_destroy`, the refcount of `dtls->sctp` is decremented and will eventually reach zero - causing the thread to end and to release the component.

Also `janus_dtls_srtp_destroy` can be called multiple times, it atomically checks and sets `dtls->destroyed`.

Reply:
:+1: